projects/packages/autoloader/src/class-php-autoloader.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/* HEADER */ // phpcs:ignore
+
+/**
+ * This class handles management of the actual PHP autoloader.
+ */
+class PHP_Autoloader {
+
+	/**
+	 * Registers the autoloader with PHP so that it can begin autoloading classes.
+	 *
+	 * @param Version_Loader $version_loader The class loader to use in the autoloader.
+	 */
+	public function register_autoloader( $version_loader ) {
+		// Make sure no other autoloaders are registered.
+		$this->unregister_autoloader();
+
+		// Set the global so that it can be used to load classes.
+		global $jetpack_autoloader_loader;
+		$jetpack_autoloader_loader = $version_loader;
+
+		// Ensure that the autoloader is first to avoid contention with others.
+		spl_autoload_register( array( self::class, 'load_class' ), true, true );
+	}
+
+	/**
+	 * Unregisters the active autoloader so that it will no longer autoload classes.
+	 */
+	public function unregister_autoloader() {
+		// Remove any v2 autoloader that we've already registered.
+		$autoload_chain = spl_autoload_functions();
+		foreach ( $autoload_chain as $autoloader ) {
+			// We can identify a v2 autoloader using the namespace.
+			$namespace_check = null;
+
+			// Functions are recorded as strings.
+			if ( is_string( $autoloader ) ) {
+				$namespace_check = $autoloader;
+			} elseif ( is_string( $autoloader[0] ) ) {
+				// Static method calls have the class as the first array element.
+				$namespace_check = $autoloader[0];
+			} else {
+				// Object method calls have the object as the first array element.
+				$namespace_check = get_class( $autoloader[0] );
```

Comment:
Since we've never registered an object autoloader I'm not sure if we would prefer to `continue;` here to avoid the potentially expensive `get_class()` function call?